db/commitlog/commitlog.cc

```diff
@@ -2800,6 +2800,8 @@ future<> db::commitlog::segment_manager::do_pending_deletes() {
     promise<> deleting_done;
     auto pending_deletes = std::exchange(_pending_deletes, deleting_done.get_future());
 
+    co_await std::move(pending_deletes);
```

Comment:
How does this achieve serialization?
Do I understand correctly that this will now cause all previous deletes to be processed, before we issue new ones below? Thus ensuring we don't decide to keep some segment we was supposed to delete?

Reply:
Yes. Before we synced _after_ doing our deletions. Now we sync before it.

Reply:
Makes sense.